.github/workflows/ci.yml

```diff
@@ -18,25 +22,100 @@ jobs:
       - name: Setup Python
         uses: actions/setup-python@v2
         with:
-          python-version: 2.7
+          python-version: 3.8
 
       - name: Install Python dependencies
         run: |
           pip install --upgrade pip
-          pip install yadage
+          pip install --pre reana-client
 
-      - name: Validate workflow
-        run: yadage-validate workflow/yadage/workflow.yaml | grep -q 'workflow validates'
+      - name: Validate workflows
+        run: |
+          for file in $FILES; do reana-client validate --environments -f $file; done
+        env:
+          FILES: reana.yaml reana-slurmcern.yaml reana-htcondorcern.yaml reana-yadage.yaml reana-yadage-slurmcern.yaml reana-yadage-htcondorcern.yaml reana-cwl.yaml reana-cwl-slurmcern.yaml reana-cwl-htcondorcern.yaml reana-snakemake.yaml
+
+
+  serial-local-run:
+    runs-on: ubuntu-20.04
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+      - name: Run Serial locally
+        uses: reanahub/reana-github-actions/local-run@v1
+        with:
+          commands: |
 
-      - run: docker run -i --rm -v `pwd`:/workdir reanahub/reana-env-root6 bash -c "
+            docker run -i --rm -v `pwd`:/workdir reanahub/reana-env-root6 bash -c "
               cd /workdir &&
               mkdir -p ./results &&
               root -b -q './code/gendata.C(20000,\"./results/data.root\")'"
-
-      - run: ls -l `pwd`/results/data.root
-
-      - run: docker run -i --rm -v `pwd`:/workdir reanahub/reana-env-root6 bash -c "
+            ls -l `pwd`/results/data.root
+            docker run -i --rm -v `pwd`:/workdir reanahub/reana-env-root6 bash -c "
               cd /workdir &&
               root -b -q './code/fitdata.C(\"./results/data.root\",\"./results/plot.png\")'"
+             ls -l `pwd`/results/plot.png
 
-      - run: ls -l `pwd`/results/plot.png
+  yadage-local-run:
+    runs-on: ubuntu-20.04
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+      - name: Run Yadage locally
+        uses: reanahub/reana-github-actions/local-run@v1
+        with:
+          commands: |
+            rm -rf yadage-local-run/yadage-inputs && mkdir -p yadage-local-run/yadage-inputs
+            cd yadage-local-run
+            virtualenv ~/.virtualenvs/yadage
+            source ~/.virtualenvs/yadage/bin/activate
```

Comment:
Maybe this code could also be moved to `local-run` in a step before [run workflow](https://github.com/reanahub/reana-github-actions/blob/master/local-run/action.yml#L21)? As well as `rm -rf yadage-local-run ~/.virtualenvs/yadage` in the end. WDYT?